docs/src/snap/howto/proxy.md

```diff
@@ -0,0 +1,31 @@
+# Configuring proxy settings for K8s
+
+Canonical Kubernetes packages a number of utilities (eg curl, helm) which need
+to fetch resources they expect to find on the internet. In a constrained
+network environment, such access is usually controlled through proxies.
+
+On Ubuntu and other Linux operating systems, proxies are configured through
+system-wide environment variables defined in the `/etc/environment` file.
+
+## Adding proxy configuration
+
+Edit the `/etc/environment` file and add the relevant URLs
+
+```{note} It is important to add whatever address ranges are used by the
+ cluster itself to the `NO_PROXY` and `no_proxy` variables.
+```
+
+For example, assume we have a proxy running at `http://squid.internal:3128` and
+we are using the networks `10.0.0.0/8`,`192.168.0.0/16` and `172.16.0.0/12`. We
```

Comment:
it could be nice to have a run-down of what these CIDRs are.  
Usually you need no-proxies for:

* service-cidr
* pod-cidr
* local network cidr

Reply:
the defaults for those in the snap are [here](https://github.com/canonical/k8s-snap/blob/af5c83997c1f98d21ced6b17a43a4b4ab8690558/src/k8s/pkg/k8sd/types/cluster_config_defaults.go#L12-L17)

Reply:
I added a not for this to both Howto guides and explained it slightly more in the reference page